pkg/cluster/cluster.go

```diff
@@ -687,6 +701,18 @@ func (c *Cluster) initInfrastructureRoles() error {
 	return nil
 }
 
+func (c *Cluster) avoidProtectedOrSystemRole(username, purpose string) bool {
```

Comment:
Probably `shouldAvoidProtectedOrSystemRole` or just `isProtectedOrSystemRole` ? I find adjectives easier to read in `if-then-else`

Reply:
shouldAvoid is good. I thought about isProtectedOrSystemRole, but the function actually emits a warning inside, so it not just gives a value to make a decision, but also suggests interprets it.